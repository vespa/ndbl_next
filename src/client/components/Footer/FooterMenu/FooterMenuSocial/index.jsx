
import React from 'react';
import PropTypes from 'prop-types';
import css from './style.scss';
import cssMain from '../../../../../scss/main.scss';
import FooterMenuSocialItem from './FooterMenuSocialItem';

const FooterMenuSocial = ({ title }) => {
  return (
    <React.Fragment>
      <h3 className={css['footer-menu__title']}>{title}</h3>
      <div className={css['footer-menu__list']}>
        <ul className={`${cssMain.col_6} ${css['footer-menu__list']}}`}>
          {[
            { text: 'Newsletter', icon: 'newsletter', href: '/' },
            { text: 'Instagram', icon: 'instagram', href: '/' },
            { text: 'Twitter', icon: 'twitter', href: '/' },
            { text: 'Tumblr', icon: 'tumblr', href: '/' },
            { text: 'Linkedin', icon: 'linkedin', href: '/' },
          ].map(item => <FooterMenuSocialItem key={item.text} {...item} />)
          }
        </ul>
        <ul className={`${cssMain.col_6} ${css['footer-menu__list']}}`}>
          {[
            { text: 'Facebook', icon: 'facebook', href: '/' },
            { text: 'Pinterest', icon: 'pinterest', href: '/' },
            { text: 'Youtube', icon: 'youtube', href: '/' },
            { text: 'Google+', icon: 'google', href: '/' },
          ].map(item => <FooterMenuSocialItem key={item.text} {...item} />)
          }
        </ul>
      </div>
    </React.Fragment>
  );
};

FooterMenuSocial.propTypes = {
  title: PropTypes.string.isRequired,
};

export default FooterMenuSocial;
