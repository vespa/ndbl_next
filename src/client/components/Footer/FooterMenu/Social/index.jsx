
import React from 'react';
import PropTypes from 'prop-types';
import css from './style.scss';
import cssMain from '../../../../../scss/main.scss';

const FooterMenuSocial = ({ title }) => {
  return (
    <React.Fragment>
      <h3 className={css['footer-menu__title']}>{title}</h3>
      <ul className={css['footer-menu__list']}>
        {/* {links.map(item => <li key={item.text}><a href={item.href}>{item.text}</a> </li>)} */}
        <ul className={cssMain.col_6}>
          teste
        </ul>
        <ul className={cssMain.col_6}>
          teste
        </ul>
      </ul>
    </React.Fragment>
  );
};

FooterMenuSocial.propTypes = {
  title: PropTypes.string.isRequired,
};

export default FooterMenuSocial;
