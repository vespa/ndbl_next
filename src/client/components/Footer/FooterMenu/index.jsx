
import React from 'react';
import PropTypes from 'prop-types';
import Link from 'next/link';
import css from './style.scss';

const FooterMenu = ({ title, links }) => {
  return (
    <React.Fragment>
      <h3 className={css['footer-menu__title']}>{title}</h3>
      <ul className={css['footer-menu__list']}>
        {links.map(item => <li key={item.text}><Link href={item.href}>{item.text}</Link> </li>)}
      </ul>
    </React.Fragment>
  );
};

FooterMenu.propTypes = {
  title: PropTypes.string.isRequired,
  links: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
};

export default FooterMenu;
