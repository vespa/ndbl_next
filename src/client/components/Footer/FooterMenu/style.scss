.footer-menu {
  &__title {
    text-transform: uppercase;
  }
  &__list {
    padding: 0;
    margin: 0;
    * {
      list-style-type: none;
      font-size: 13px;
    }
    a {
      color: #aaa;
      display: block;
      margin-bottom: 10px;
      line-height: 16px;
    }
  }
}