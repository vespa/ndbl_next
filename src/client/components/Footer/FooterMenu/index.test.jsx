import React from 'react';
import { shallow } from 'enzyme';
import FooterMenu from './index';

describe('[Component: FooterMenu]', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<FooterMenu title="test" links={[{ text: 'Home', href: '/' }]} />);
  });
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
