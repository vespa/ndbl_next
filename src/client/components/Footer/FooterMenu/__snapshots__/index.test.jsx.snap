// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component: FooterMenu] should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FooterMenu
    links={
      Array [
        Object {
          "href": "/",
          "text": "Home",
        },
      ]
    }
    title="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h3
          className={undefined}
        >
          test
        </h3>,
        <ul
          className={undefined}
        >
          <li>
            <Link
              href="/"
            >
              Home
            </Link>
             
          </li>
        </ul>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
          "className": undefined,
        },
        "ref": null,
        "rendered": "test",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <Link
                href="/"
              >
                Home
              </Link>
               
            </li>,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "Home",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  href="/"
                >
                  Home
                </Link>,
                " ",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Home",
                  "href": "/",
                },
                "ref": null,
                "rendered": "Home",
                "type": [Function],
              },
              " ",
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h3
            className={undefined}
          >
            test
          </h3>,
          <ul
            className={undefined}
          >
            <li>
              <Link
                href="/"
              >
                Home
              </Link>
               
            </li>
          </ul>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test",
            "className": undefined,
          },
          "ref": null,
          "rendered": "test",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <Link
                  href="/"
                >
                  Home
                </Link>
                 
              </li>,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "Home",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Link
                    href="/"
                  >
                    Home
                  </Link>,
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Home",
                    "href": "/",
                  },
                  "ref": null,
                  "rendered": "Home",
                  "type": [Function],
                },
                " ",
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
